src/client/utils/ajax/ajax.ts

```diff
@@ -62,6 +64,24 @@ export class Ajax {
       });
   }
 
+  static newQuery(essence: Essence, { clientTimeout: timeout, oauth }: ClientAppSettings): Promise<Dataset> {
```

Comment:
Probably we could partially apply this function at the start of client application (we would eliminate passing appSettings every time) and distribute it with context. DataProvider would consume it from there.